### DISCUSSION TITLE
Help sending docx for pdf conversion as bytes - python

### DISCUSSION THREAD

**Discussion by tayiabscode** [gotenberg/gotenberg#580] [Q&A] [answered]
Hi there,

I recently discovered gotenberg and wanted to ask whether it is possible to send the .docx file as bytes for conversion as opposed to supplying a file path. For my potential use-case files cannot be stored locally/cannot persist anywhere. I am using the following python code to make the request to the gottenberg docker image:

```
import requests
from io import BytesIO

with open('worddoc.docx','rb') as f:
    bytes=f.read()
file_stream = BytesIO(bytes)

file={'file':file_stream}
r=requests.post(url='http://localhost:3000/forms/libreoffice/convert',files=file)
print(r.content)
```

but receive the following result:

`b'Invalid form data: no form file found for extensions: [.bib .doc .xml .docx .fodt .html .ltx .txt .odt .ott .pdb .pdf .psw .rtf .sdw .stw .sxw .uot .vor .wps .epub .png .bmp .emf .eps .fodg .gif .jpg .jpeg .met .odd .otg .pbm .pct .pgm .ppm .ras .std .svg .svm .swf .sxd .sxw .tif .tiff .xhtml .xpm .odp .fodp .potm .pot .pptx .pps .ppt .pwp .sda .sdd .sti .sxi .uop .wmf .csv .dbf .dif .fods .ods .ots .pxl .sdc .slk .stc .sxc .uos .xls .xlt .xlsx]'`

I did see [this post](https://github.com/gotenberg/gotenberg/discussions/481) that seems to confirm it is possible to send as bytes in C# but I'm not sure how it can be done in Python or where I am going wrong?

Any help would be appreciated :)

**Comment by gulien** [comment, marked as answer]
Hello @tayiabscode!

The LibreOffice convert endpoint accepts `multipart/form-data` requests. In your case, you should make sure the filename associated with the form file bytes content ends with `.docx`.

**Reply by tayiabscode** [reply]
Thanks @gulien I misunderstood previously but for future reference simply had to ensure the name in the `file` dictionary also ends in `.docx`
